lib/bgpstream_di_mgr.c

```diff
@@ -358,11 +365,36 @@ int bgpstream_di_mgr_get_next_record(bgpstream_di_mgr_t *di_mgr,
   int rc;
 
   while (1) {
-    // if our queue is empty, ask the DI for more
-    if (bgpstream_resource_mgr_empty(di_mgr->res_mgr) != 0 &&
-        ACTIVE_DI->update_resources(ACTIVE_DI) != 0) {
-      // an error occurred
-      return -1;
+    // if our queue is empty, or we only have stream resources and the
+    // poll timer has expired, then ask the DI for more resources
+    if (bgpstream_resource_mgr_empty(di_mgr->res_mgr) != 0 ||
+        (bgpstream_resource_mgr_stream_only(di_mgr->res_mgr) != 0 &&
+         epoch_sec() >= di_mgr->next_poll)) {
+
+      bgpstream_log(BGPSTREAM_LOG_WARN, "asking DI for resources");
```

Comment:
this message  will show up at the beginning of each run of `bgpreader`. I don't think it convey much information, especially as a warning. I'd vote removing this or enclose this in a `ifdef`.

Reply:
oh yeah, i don’t think that should be there at all